packages/cli/src/PublicApi/v1/handlers/workflows/workflows.service.ts

```diff
@@ -79,20 +72,21 @@ export async function createWorkflow(
 	user: User,
 	role: Role,
 ): Promise<WorkflowEntity> {
-	let savedWorkflow: unknown;
-	const newWorkflow = new WorkflowEntity();
-	Object.assign(newWorkflow, workflow);
-	await Db.transaction(async (transactionManager) => {
-		savedWorkflow = await transactionManager.save<WorkflowEntity>(newWorkflow);
+	return Db.transaction(async (transactionManager) => {
+		const newWorkflow = new WorkflowEntity();
+		Object.assign(newWorkflow, workflow);
```

Comment:
in this case fine, however I would usually suggest to keep transactions as lean as possible since they are blocking on sqlite